protocol/communities/manager.go

```diff
@@ -4605,43 +4605,26 @@ func (m *Manager) HandleCommunityTokensMetadata(communityID string, communityTok
 					Decimals:           int(tokenMetadata.Decimals),
 				}
 
-				callOpts := &bind.CallOpts{Context: context.Background(), Pending: false}
-
 				switch tokenMetadata.TokenType {
 				case protobuf.CommunityTokenType_ERC721:
-					contract, err := m.collectiblesService.API().GetCollectiblesContractInstance(chainID, address)
-					if err != nil {
-						return err
-					}
-					totalSupply, err := contract.TotalSupply(callOpts)
-					if err != nil {
-						return err
-					}
-					transferable, err := contract.Transferable(callOpts)
-					if err != nil {
-						return err
-					}
-					remoteBurnable, err := contract.RemoteBurnable(callOpts)
+					contractData, err := m.collectiblesService.GetCollectibleContractData(chainID, address)
 					if err != nil {
 						return err
 					}
 
-					communityToken.Supply = &bigint.BigInt{Int: totalSupply}
-					communityToken.Transferable = transferable
-					communityToken.RemoteSelfDestruct = remoteBurnable
-					communityToken.InfiniteSupply = collectibles.GetInfiniteSupply().Cmp(totalSupply) == 0
+					communityToken.Supply = contractData.TotalSupply
+					communityToken.Transferable = contractData.Transferable
+					communityToken.RemoteSelfDestruct = contractData.RemoteBurnable
+					communityToken.InfiniteSupply = contractData.InfiniteSupply
 
 				case protobuf.CommunityTokenType_ERC20:
-					contract, err := m.collectiblesService.API().GetAssetContractInstance(chainID, address)
-					if err != nil {
-						return err
-					}
-					totalSupply, err := contract.TotalSupply(callOpts)
+					contractData, err := m.collectiblesService.GetAssetContractData(chainID, address)
 					if err != nil {
 						return err
 					}
-					communityToken.Supply = &bigint.BigInt{Int: totalSupply}
-					communityToken.InfiniteSupply = collectibles.GetInfiniteSupply().Cmp(totalSupply) == 0
+
+					communityToken.Supply = contractData.TotalSupply
+					communityToken.InfiniteSupply = contractData.InfiniteSupply
 				}
 
 				err = m.persistence.AddCommunityToken(communityToken)
```

Comment:
We miss 2 fields here: PrivilegeLevel and Deployer. It will be hard to get deployer since we need to parse blockchain transactions. PrivilegeLevel can be taken from permissions.